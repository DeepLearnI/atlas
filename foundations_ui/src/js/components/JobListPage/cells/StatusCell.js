import React, { Component } from 'react';
import PropTypes from 'prop-types';
import JobListActions from '../../../actions/JobListActions';

class StatusCell extends Component {
  constructor(props) {
    super(props);
    this.state = {
      status: this.props.status,
      isError: this.props.isError,
      rowNumber: this.props.rowNumber,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.rowNumber !== this.props.status) {
      this.setState({
        status: nextProps.status,
        isError: nextProps.isError,
        rowNumber: nextProps.rowNumber,
      });
    }
  }

  render() {
    const { status, isError, rowNumber } = this.state;

    const statusClass = JobListActions.getStatusCircle(status);
    const divClass = isError ? `status-cell error row-${rowNumber}` : `status-cell  row-${rowNumber}`;

    return (
      <span className={divClass} data-class={`job-status-${status}`}>
        { statusClass.includes('status-running')
          ? (
            <svg className="spinner" width="18px" height="18px" viewBox="0 0 66 66" xmlns="http://www.w3.org/2000/svg">
              <circle
                className="path"
                fill="none"
                strokeWidth="6"
                strokeLinecap="round"
                cx="33"
                cy="33"
                r="30"
              />
            </svg>
          )
          : <span className={statusClass} />}
      </span>
    );
  }
}

StatusCell.propTypes = {
  status: PropTypes.string,
  isError: PropTypes.bool,
  rowNumber: PropTypes.number,
};

StatusCell.defaultProps = {
  status: '',
  isError: false,
  rowNumber: 0,
};

export default StatusCell;
