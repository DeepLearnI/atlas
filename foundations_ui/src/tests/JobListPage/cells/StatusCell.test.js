import React from 'react';
import ReactDOM from 'react-dom';
import StatusCell from '../../../js/components/JobListPage/cells/StatusCell';
import { shallow } from 'enzyme';
import configureTests from '../../setupTests';

configureTests();

it('Shallow Renders Status Cell', () => {
  const wrapper = shallow(<StatusCell/>);
  expect(wrapper).toMatchSnapshot();
});