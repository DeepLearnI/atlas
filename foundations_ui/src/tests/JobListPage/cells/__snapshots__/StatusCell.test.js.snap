// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Status Cell 1`] = `
<span
  className="status-cell  row-0"
>
  <span
    className="status-icon status-green"
  />
</span>
`;
